scipy/linalg/matfuncs.py

```diff
@@ -668,3 +675,329 @@ def rounded_sign(x):
         return S0
     else:
         return S0, errest
+
+
+def is_matrix_hermitian(A):
+    """
+    Returns True if a matrix is hermitian
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is hermitian.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_hermitian
+    >>> m = np.array([[0,1j],[-1j,0]])
+    >>> print(is_matrix_hermitian(m))
+    True
+    >>> m = np.array([[5,0],[1,3]])
+    >>> print(is_matrix_hermitian(m))
+    False
+    """
+    A = _asarray_square(A)
+    return np.sum(A == A.conj().T) == (A.shape[1]**2)
+
+
+def is_matrix_symmetric(A):
+    """
+    Returns True if a matrix is symmetric
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is symmetric.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_symmetric
+    >>> m = np.array([[2,3],[3,2]])
+    >>> print(is_matrix_symmetric(m))
+    True
+    >>> m = np.array([[5,0],[1,3]])
+    >>> print(is_matrix_symmetric(m))
+    False
+    """
+    A = _asarray_square(A)
+    return np.sum(A == A.T) == (A.shape[1]**2)
+
+
+def is_matrix_skew_symmetric(A, tol=1e-8):
```

Comment:
`is_skew_symmetric()`. Also default tol shouldn't be hard-coded as I mentioned in the other PR. It has to be derived from the inf norm of the matrix. See other tolerances in `basic.py`.

Reply:
All tolerances are now calculated as you suggested.